src/crypto/sha256_power8.cpp

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+//
+// This is a translation to GCC extended asm syntax from YASM code by Intel
+// (available at the bottom of this file).
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#ifdef HAVE_POWER8
+
+#include <stdint.h>
+
+#if defined(HAVE_ENDIAN_H)
+#include <endian.h>
+#elif defined(HAVE_SYS_ENDIAN_H)
+#include <sys/endian.h>
+#endif
+
+#include <altivec.h>
+
+namespace sha256_power8
+{
+
+typedef __vector uint32_t uint32x4_p8;
+typedef __vector uint8_t uint8x16_p8;
```

Comment:
`__vector` is a non-standard GCC extension; any reason we can't use `vector` here directly?